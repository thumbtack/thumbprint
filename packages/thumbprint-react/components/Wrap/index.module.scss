@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';
@import '~@thumbtack/thumbprint-scss/mixins';

.root {
    margin-left: auto;
    margin-right: auto;
    width: 100%;
    max-width: $tp-wrap__max-width;
    padding-left: $tp-space__3;
    padding-right: $tp-space__3;

    @include tp-respond-above($tp-breakpoint__small) {
        padding-left: $tp-space__5;
        padding-right: $tp-space__5;
    }

    @include tp-respond-above($tp-wrap__no-pad-width) {
        padding-left: 0;
        padding-right: 0;
    }
}

.bleedBelowSmall {
    @include tp-respond-below($tp-breakpoint__small) {
        padding-left: 0;
        padding-right: 0;
    }
}

.bleedBelowMedium {
    @include tp-respond-below($tp-breakpoint__medium) {
        padding-left: 0;
        padding-right: 0;
    }
}

.bleedBelowLarge {
    @include tp-respond-below($tp-breakpoint__large) {
        padding-left: 0;
        padding-right: 0;
    }
}
