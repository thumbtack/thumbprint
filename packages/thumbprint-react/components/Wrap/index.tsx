import React from 'react';
import classNames from 'classnames';
import styles from './index.module.scss';

interface PropTypes {
    /**
     * Component to be wrapped.
     */
    children?: React.ReactNode;
    /**
     * Breakpoint at which the wrapper should remove horizontal margin to bleed to the edge of the
     * viewport. Defaults to `undefined`, in which case it never does this.
     */
    bleedBelow?: 'small' | 'medium' | 'large';
    /**
     * A selector hook into the React component for use in automated testing environments.
     */
    dataTest?: string;
}

export default function Wrap({ children, bleedBelow, dataTest }: PropTypes): JSX.Element {
    return (
        <div
            className={classNames({
                [styles.root]: true,
                [styles.bleedBelowSmall]: bleedBelow === 'small',
                [styles.bleedBelowMedium]: bleedBelow === 'medium',
                [styles.bleedBelowLarge]: bleedBelow === 'large',
            })}
            data-test={dataTest}
        >
            {children}
        </div>
    );
}
