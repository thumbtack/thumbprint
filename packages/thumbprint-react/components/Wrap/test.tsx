import React from 'react';
import { mount } from 'enzyme';
import Wrap from './index';

test('renders content that is passed in', () => {
    const wrapper = mount(<Wrap>goose</Wrap>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});

test('renders content that is passed in with a bleed breakpoint', () => {
    const wrapper = mount(<Wrap bleedBelow="medium">goose</Wrap>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});

test('adds `dataTest` prop', () => {
    const wrapper = mount(<Wrap dataTest="Duck">goose</Wrap>);
    expect(wrapper.find('div[data-test="Duck"]')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});
