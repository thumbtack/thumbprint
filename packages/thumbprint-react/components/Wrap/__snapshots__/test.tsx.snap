// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds \`dataTest\` prop 1`] = `
<Wrap
  dataTest="Duck"
>
  <div
    className="root"
    data-test="Duck"
  >
    goose
  </div>
</Wrap>
`;

exports[`renders content that is passed in 1`] = `
<Wrap>
  <div
    className="root"
  >
    goose
  </div>
</Wrap>
`;

exports[`renders content that is passed in with a bleed breakpoint 1`] = `
<Wrap
  bleedBelow="medium"
>
  <div
    className="root bleedBelowMedium"
  >
    goose
  </div>
</Wrap>
`;
